### PR TITLE
Mange etcd with etcd-operator

### LINKED ISSUE #307: Utilize etcd-operator

### PR DESCRIPTION
**Pull request by scheeles**
closes #307

### CODE REVIEW

File: extensions/etcd/client.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+)
+
+// ConstructNewSerialKeyName generates a name for a serial key which is accepted by k8s metadata.Name
+// Fingerprint is without colons
+func ConstructNewSerialKeyName(fingerprint string) string {
+	return fmt.Sprintf("key-%s-%s", fingerprint, uuid.ShortUID(4))
+}
+
+// NormalizeFingerprint returns a normalized fingerprint
+func NormalizeFingerprint(f string) string {
```
Comment (mrIncompetent) [inline extensions/etcd/client.go]: Needed?

File: extensions/etcd/client.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/runtime/schema"
+	"k8s.io/client-go/pkg/runtime/serializer"
+	"k8s.io/client-go/pkg/watch"
+	"k8s.io/client-go/rest"
+)
+
+// ConstructNewSerialKeyName generates a name for a serial key which is accepted by k8s metadata.Name
+// Fingerprint is without colons
+func ConstructNewSerialKeyName(fingerprint string) string {
```
Comment (mrIncompetent) [inline extensions/etcd/client.go]: Needed?

File: extensions/etcd/client.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+	return fmt.Sprintf("key-%s-%s", fingerprint, uuid.ShortUID(4))
+}
+
+// NormalizeFingerprint returns a normalized fingerprint
+func NormalizeFingerprint(f string) string {
+	return strings.NewReplacer(":", "").Replace(f)
+}
+
+// NormalizeUser base64 encodes a user to store him in labels
+func NormalizeUser(name string) string {
```
Comment (mrIncompetent) [inline extensions/etcd/client.go]: Needed?

File: extensions/etcd/client.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+	return base64.URLEncoding.WithPadding(base64.NoPadding).EncodeToString([]byte(name))
+}
+
+// GenerateNormalizedFigerprint a normalized fingerprint from a public key
+func GenerateNormalizedFigerprint(pub string) (string, error) {
```
Comment (mrIncompetent) [inline extensions/etcd/client.go]: Needed?

File: controller/cluster/pending.go [hunk 5]
```diff
@@ -352,7 +358,8 @@ func (cc *clusterController) launchingCheckPvcs(c *api.Cluster) error {
 	ns := kubernetes.NamespaceName(c.Metadata.User, c.Metadata.Name)
 
 	pvcs := map[string]func(c *api.Cluster, app, masterResourcesPath string) (*v1.PersistentVolumeClaim, error){
-		"etcd": resources.LoadPVCFile,
+	// Currently not required pvc for etcd is done by etcd-operator
```
Comment (mrIncompetent) [inline controller/cluster/pending.go:361]: Then just remove the line

File: controller/resources/load_files.go [hunk 6]
```diff
@@ -135,3 +136,27 @@ disablestrictzonecheck=true`, c.Spec.Cloud.AWS.AvailabilityZone, c.Spec.Cloud.AW
 	return &cm, nil
 
 }
+
+// LoadEtcdClustertFile loads a etcd-operator tpr from disk and returns a Cluster tpr struct
+func LoadEtcdClustertFile(c *api.Cluster, v *api.MasterVersion, masterResourcesPath, dc, yamlFile string) (*etcd.Cluster, error) {
+
+	data := struct {
+		DC               string
+		AdvertiseAddress string
```
Comment (mrIncompetent) [inline controller/resources/load_files.go]: Should be removed if not used

**Review by mrIncompetent — CHANGES_REQUESTED** [review]

File: controller/cluster/pending.go (hunk 5 above)
Comment (scheeles) [reply controller/cluster/pending.go:361]: Don't want to remove it now, because if we don't need PVC in the future we can remove it

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**